sound/soc/sof/core.c

```diff
@@ -346,9 +352,12 @@ static int sof_probe(struct platform_device *pdev)
 	snd_soc_unregister_component(&pdev->dev);
 	snd_sof_free_topology(sdev);
 fw_run_err:
+/* bypass DSP if in force legacy hda debug mode */
+#ifndef CONFIG_SND_SOC_SOF_FORCE_LEGACY_HDA
 	snd_sof_fw_unload(sdev);
 fw_load_err:
 	snd_sof_ipc_free(sdev);
+#endif
 ipc_err:
 	snd_sof_free_debug(sdev);
```

Comment:
isn't the debug tied to DSP?

Reply:
I need refine this part, and adding check to dsp availability also.